ENYO-2579: Add workaround for preventing multi-tap issue

kbs12e opened the pull request:
## Issue

The focus does not released when the list type components are multi-tapped
## Cause

DataList, List and GridList item has no workaroud (ongesturechange, ongestureend, onleave) for preventing remained pressed state.
## Fix

Add ongesturechange and ongestureend event to DataGridList for preventing multi-tap issue. However,In List and GridList,  this workaround(ongesturechange, ongestureend, onleave) is not fired because of flyweight pattern 's limitation, so we add ondragout to List type component instead of onleave.
In addition, we found that List item is remained press state after onhold event, so we added onrelease event for preventing it.

Enyo-DCO-1.1-Signed-off-by: Bongsub Kim bongsub.kim@lgepartner.com

kbs12e committed:
ENYO-2579: Add workaround for preventing multi-tap issue

https://jira2.lgsvl.com/browse/ENYO-2128
Enyo-DCO-1.1-Signed-off-by: Bongsub Kim <bongsub.kim@lgepartner.com>

kbs12e committed:
ENYO-2579: Modify List type sample for preventing remained pressed state

https://jira2.lgsvl.com/browse/ENYO-2579
Enyo-DCO-1.1-Signed-off-by: Bongsub Kim <bongsub.kim@lgepartner.com>

Change-Id: I6eb2ef8bda89ffd640d8cacb904974c899b93562

seunghoh commented:
Found side effect. 
Tab&Hold on ListItem > ListItem status changed to Drag&Drop > Release tap > All Itmes are selected

kbs12e committed:
ENYO-2579: Add onrelease event for preventing remained press state after onhold event

https://jira2.lgsvl.com/browse/ENYO-2579
Enyo-DCO-1.1-Signed-off-by: Bongsub Kim <bongsub.kim@lgepartner.com>

Change-Id: Ib02a58a82bbc530693689b1d184851d17510dbae
